### ISSUE TITLE
Bug in SearchALinkedListWithJumps

### ISSUE THREAD

**Issue by chellitorvic** [bephrem1/backtobackswe#23]
While boxed type Integer is stored in a heap, it is still immutable. You can see in the Integer class sources that the primitive value it wraps is declared as `final` [here](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/Integer.java#l840). 

Another thing is that objects are indeed passed to methods as references, but that references are copied to a local variable (parameter) of a method. 
So `currentOrder += 1` is the same as `currentOrder  = currentOrder + 1`, and JVM, in that case, unboxes `currentOrder` to a primitive value, performs addition and then boxes result into a new instance of Integer. Then the reference to this new object is assigned to `currentOrder`, overriding the initial value of it. But it held just a copy of a reference passed, so the original variable used as an argument remains the same, pointing to an old boxing object instance.

This can be observed from a simple example [here](https://repl.it/repls/MemorableSlateblueCustomization) or by using the following list to run the program:
```java
ListNode a = new ListNode();
ListNode b = new ListNode();
ListNode c = new ListNode();
a.next = b;
b.next = c;
a.jump = c;
```
Result should be `a:0 b:2 c:1`, but it is `a:0 b:1 c:1`.

**Comment by chellitorvic** [comment]
Fixed in pull request #25

**Comment by bephrem1** [comment]
Oh good catch ha, I goofed
